src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -181,41 +174,34 @@ public void updateFilteredCustomerList(Predicate<Customer> predicate) {
 
     //=========== Filtered Commission List Accessors =============================================================
 
-    /**
-     * Returns an unmodifiable view of the list of {@code Commission} backed by the internal list of
-     * {@code versionedAddressBook}
-     */
     @Override
-    public ObservableList<Commission> getFilteredCommissionList() {
-        return filteredCommissions;
+    public FilteredList<Commission> getFilteredCommissionList() {
+        return observableFilteredCommissions.getValue();
     }
 
     @Override
-    public void updateFilteredCommissionList(Predicate<Commission> predicate) {
-        requireNonNull(predicate);
-        filteredCommissions.setPredicate(predicate);
-    }
-
-    private Predicate<Commission> getSelectedCustomerCommissionsPredicate() {
-        return commission -> commission.getCustomer().isSameCustomer(getSelectedCustomer().getValue());
+    public ObservableValue<FilteredList<Commission>> getObservableFilteredCommissionList() {
+        return observableFilteredCommissions;
     }
 
-    public void updateFilteredCommissionListToSelectedCustomer() {
-        updateFilteredCommissionList(getSelectedCustomerCommissionsPredicate());
+    @Override
+    public void updateFilteredCommissionList(Predicate<Commission> predicate) {
+        requireAllNonNull(observableFilteredCommissions.getValue(), predicate);
+        observableFilteredCommissions.getValue().setPredicate(predicate);
     }
 
     //=========== Selected Customer =============================================================
 
     @Override
     public void selectCustomer(Customer customer) {
-        selectedCommission.setValue(null); // resets the selected commission
```

Comment:
Also, just to clarify- by removing the reset of selectedCommission each time a Customer is selected, this means that it is possible for the selectedCommission not to belong to the selectedCustomer at a given point of time. 

While I don't think this causes any bugs, I'm just wondering if this would be confusing for the user because if I'm not wrong, it's now actually possible to see and add/ delete iterations from a 'deleted commission' if you:
1. open a commission from a customer,
2. switch back to the customer tab and select a different customer
3. now delete the initial customer selected
4. notice that the selected commission is still the commission from the now deleted customer, and since its a composition, this commission is also 'deleted'- but it's still possible to run `additer` and `deliter` commands on this deleted commission

Do you think this might be an area of concern? Or is it a non-issue/ intended behaviour?

Reply:
Other than this question, the rest of the code looks good. Thanks for all the hard work! Let's merge this in ASAP so you (hopefully) don't have to deal with any more merge conflicts.

Reply:
Yes you are right, that line shouldn't have been deleted.